File: 프로그래머스/2/12914. 멀리 뛰기/멀리 뛰기.js
function solution(n) {
    
    const dp = [];
    
    dp.push(0);
    dp.push(1);
    dp.push(2);
    
    for(let i=3; i<n+1; i++){
        dp.push((dp[i-1]+dp[i-2])%1234567);
    }

    return dp[n];
}

File: 프로그래머스/2/12914. 멀리 뛰기/멀리 뛰기.java
class Solution {
    public long solution(int n) {
        
        long dp[];
    
        if(n < 3){
            dp = new long[3];
        }else{
            dp = new long[n+1];
        }        
        
        dp[1] = 1;
        dp[2] = 2;
        
        for(int i=3; i<=n; i++){
            dp[i] = (dp[i-2] + dp[i-1])%1234567;
        }
        
        return dp[n];
    }
}